Src/CSharpier.Rider/src/main/java/com/intellij/csharpier/ProcessHelper.java

```diff
@@ -47,4 +57,32 @@ public static String ExecuteCommand(String[] command, Map<String, String> env, F
         }
         return null;
     }
+
+    private static String GetPathWithDotNetBinary() {
+        var path = System.getenv("PATH");
+
+        // For Mac, the .NET binary isn't available for ProcessBuilder, so we'll add the default
+        // installation location to the PATH. We'll prefer the ARM version and fallback to the x64.
+        if (SystemUtils.IS_OS_MAC) {
+            return path + ":/usr/local/share/dotnet:/usr/local/share/dotnet/x64/dotnet";
+        }
+
+        // For others, it seems the .NET binary is already available to ProcessBuilder by default.
+        // So we'll just return the PATH as is.
+        return path;
+    }
+
+    // Setting the PATH doesn't affect the ProcessBuilder's command, but it will apply to
+    // spawned processes, so we'll run the desired command in the OS's default shell.
+    private static String[] GetShellCommand(String[] command) {
+        if (SystemUtils.IS_OS_MAC) {
+            return new String[] {"/bin/zsh", "-c", String.join(" ", command)};
+        }
+
+        if (SystemUtils.IS_OS_WINDOWS) {
```

Comment:
Seems a little overkill to launch a shell for all platforms, when the workaround is only needed for macos.
How about renaming this method to `GetShellCommandIfNeeded()` and make it just return the original command if it's not currently MacOS.

Reply:
Sure thing. I was aiming for normalizing the way we run commands but totally understand your perspective. I'll make that change and push it up shortly.